server/src/test/java/io/druid/client/CachingClusteredClientTest.java

```diff
@@ -469,6 +523,164 @@ public void testTimeseriesCaching() throws Exception
+          {
+            return serverView.getQueryRunner(server);
+          }
+
+          @Override
+          public void registerServerCallback(Executor exec, ServerCallback callback)
+          {
+
+          }
+        },
+        cache,
+        jsonMapper,
+        MoreExecutors.sameThreadExecutor(),
+        new CacheConfig()
+        {
+          @Override
+          public boolean isPopulateCache()
+          {
+            return true;
+          }
+
+          @Override
+          public boolean isUseCache()
+          {
+            return true;
+          }
+
+          @Override
+          public boolean isQueryCacheable(Query query)
+          {
+            return true;
+          }
+
+          @Override
+          public int getCacheBulkMergeLimit()
+          {
+            return 10;
+          }
+        }
+    );
+    final DruidServer lastServer = servers[random.nextInt(servers.length)];
+    final DataSegment dataSegment = EasyMock.createNiceMock(DataSegment.class);
+    EasyMock.expect(dataSegment.getIdentifier()).andReturn(DATA_SOURCE).anyTimes();
+    EasyMock.replay(dataSegment);
+    final ServerSelector selector = new ServerSelector(
+        dataSegment,
+        new HighestPriorityTierSelectorStrategy(new RandomServerSelectorStrategy())
+    );
+    selector.addServer(new QueryableDruidServer(lastServer, null));
+    timeline.add(interval, "v", new SingleElementPartitionChunk<>(selector));
+
+    client.run(query, context);
+
+    EasyMock.verify(cache);
+
+    EasyMock.reset(cache);
+    EasyMock.replay(cache);
+    client = new CachingClusteredClient(
+        WAREHOUSE,
+        new TimelineServerView()
+        {
+          @Override
+          public void registerSegmentCallback(Executor exec, SegmentCallback callback)
+          {
+          }
+
+          @Override
+          public VersionedIntervalTimeline<String, ServerSelector> getTimeline(DataSource dataSource)
+          {
+            return timeline;
+          }
+
+          @Override
+          public <T> QueryRunner<T> getQueryRunner(DruidServer server)
+          {
+            return serverView.getQueryRunner(server);
+          }
+
+          @Override
+          public void registerServerCallback(Executor exec, ServerCallback callback)
+          {
+
+          }
+        },
+        cache,
+        jsonMapper,
+        MoreExecutors.sameThreadExecutor(),
+        new CacheConfig()
+        {
+          @Override
+          public boolean isPopulateCache()
+          {
+            return true;
+          }
+
+          @Override
+          public boolean isUseCache()
+          {
+            return true;
+          }
+
+          @Override
+          public boolean isQueryCacheable(Query query)
+          {
+            return true;
+          }
+
+          @Override
+          public int getCacheBulkMergeLimit()
+          {
+            return 0; // <----- This is the change
+          }
+        }
+    );
```

Comment:
nit: is it possible to refactor CachingClusteredClient construction into a private method with appropriate args so that we don't have to copy paste same in various places.

LGTM otherwise